.github/alpine_32bit_log_warnings

```diff
@@ -0,0 +1,101 @@
+../../gcc/gcc.cc:2432:30: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2433:32: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2458:30: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2459:32: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2467:30: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2468:32: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2477:30: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2478:32: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2487:30: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2488:32: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2527:26: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2527:59: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2539:22: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2539:51: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/gcc.cc:2553:22: warning: too many arguments for format [-Wformat-extra-args]
+../../gcc/gcc.cc:2553:51: warning: unknown conversion type character 't' in format [-Wformat=]
+../../gcc/text-art/style.cc:150:25: warning: spurious leading punctuation sequence ';' in format [-Wformat-diag]
+../../gcc/text-art/style.cc:160:25: warning: spurious leading punctuation sequence ';' in format [-Wformat-diag]
+../../gcc/text-art/table.cc:981:62: warning: unquoted keyword 'char' in format [-Wformat-diag]
+../../gcc/text-art/table.cc:981:69: warning: spurious trailing punctuation sequence '])' in format [-Wformat-diag]
+/usr/include/c++/13.2.1/bits/new_allocator.h:172:33: warning: '*(std::_Vector_base<unsigned int, std::allocator<unsigned int> >*)((char*)&saved + offsetof(Rust::BIR::PatternBindingBuilder::SavedState, Rust::BIR::PatternBindingBuilder::SavedState::regions.tl::optional<Rust::FreeRegions>::<unnamed>.tl::detail::optional_move_assign_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_copy_assign_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_move_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_copy_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_operations_base<Rust::FreeRegions>::<unnamed>.tl::detail::optional_storage_base<Rust::FreeRegions, false>::<unnamed>)).std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_M_impl.std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_Vector_impl::<anonymous>.std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_Vector_impl_data::_M_start' may be used uninitialized [-Wmaybe-uninitialized]
```

Comment:
there are a couple of warnings like this one which seem to spruce from our code even though the warning seems to come form a std header. we should either fix them now or open up an issue and take care of them later

Reply:
i think i'd opt to open up an issue for it to be fixed later. although i thought our code is only from gcc/rust and gcc/testsuite, is text-art/ also involved?

Reply:
@badumbatish Note that the list here is, I think, from `grep warning:`, which means it misses the context lines GCC shows. It might be that e.g. there's a template in `text-art` which `gcc/rust` instantiates.

Reply:
i'll dig up the build log but it's just complaining about formatting issues, quite weird. In one instance it's giving a warning about a comment

Reply:
text-art/style.cc:160
```
2024-08-16T10:34:27.2120743Z make[2]: Entering directory '/home/runner/work/gccrs/gccrs/gccrs-build/gcc'
2024-08-16T10:34:27.2129865Z i586-alpine-linux-musl-g++  -fno-PIE -c   -g -O2     -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual  -fno-common  -DHAVE_CONFIG_H -fno-PIE -I. -Itext-art -I../../gcc -I../../gcc/text-art -I../../gcc/../include -I ./../gettext/intl -I../../gcc/../libcpp/include -I../../gcc/../libcody -I/home/runner/work/gccrs/gccrs/gccrs-build/./gmp -I/home/runner/work/gccrs/gccrs/gmp -I/home/runner/work/gccrs/gccrs/gccrs-build/./mpfr/src -I/home/runner/work/gccrs/gccrs/mpfr/src -I/home/runner/work/gccrs/gccrs/mpc/src  -I../../gcc/../libdecnumber -I../../gcc/../libdecnumber/bid -I../libdecnumber -I../../gcc/../libbacktrace -I/home/runner/work/gccrs/gccrs/gccrs-build/./isl/include -I/home/runner/work/gccrs/gccrs/isl/include  -o text-art/style.o -MT text-art/style.o -MMD -MP -MF text-art/.deps/style.TPo ../../gcc/text-art/style.cc
2024-08-16T10:34:27.2138466Z ../../gcc/text-art/style.cc: In member function 'void text_art::style::color::print_sgr(pretty_printer*, bool, bool&) const':
2024-08-16T10:34:27.2140554Z ../../gcc/text-art/style.cc:150:25: warning: spurious leading punctuation sequence ';' in format [-Wformat-diag]
2024-08-16T10:34:27.2145045Z   150 |         pp_printf (pp, ";5;%i", (int)u.m_8bit);
2024-08-16T10:34:27.2146093Z       |                         ^
2024-08-16T10:34:27.2147619Z ../../gcc/text-art/style.cc:160:25: warning: spurious leading punctuation sequence ';' in format [-Wformat-diag]
2024-08-16T10:34:27.2149088Z   160 |         pp_printf (pp, ";2;%i;%i;%i",
2024-08-16T10:34:27.2149894Z       |                         ^
2024-08-16T10:34:27.2151028Z make[2]: Leaving directory '/home/runner/work/gccrs/gccrs/gccrs-build/gcc'
```
text-art/table.cc:981
```
2024-08-16T10:34:31.2073593Z make[2]: Entering directory '/home/runner/work/gccrs/gccrs/gccrs-build/gcc'
2024-08-16T10:34:31.2083000Z i586-alpine-linux-musl-g++  -fno-PIE -c   -g -O2     -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual  -fno-common  -DHAVE_CONFIG_H -fno-PIE -I. -Itext-art -I../../gcc -I../../gcc/text-art -I../../gcc/../include -I ./../gettext/intl -I../../gcc/../libcpp/include -I../../gcc/../libcody -I/home/runner/work/gccrs/gccrs/gccrs-build/./gmp -I/home/runner/work/gccrs/gccrs/gmp -I/home/runner/work/gccrs/gccrs/gccrs-build/./mpfr/src -I/home/runner/work/gccrs/gccrs/mpfr/src -I/home/runner/work/gccrs/gccrs/mpc/src  -I../../gcc/../libdecnumber -I../../gcc/../libdecnumber/bid -I../libdecnumber -I../../gcc/../libbacktrace -I/home/runner/work/gccrs/gccrs/gccrs-build/./isl/include -I/home/runner/work/gccrs/gccrs/isl/include  -o text-art/table.o -MT text-art/table.o -MMD -MP -MF text-art/.deps/table.TPo ../../gcc/text-art/table.cc
2024-08-16T10:34:31.2091207Z ../../gcc/text-art/table.cc: In function 'void selftest::test_spans_3()':
2024-08-16T10:34:31.2092701Z ../../gcc/text-art/table.cc:981:62: warning: unquoted keyword 'char' in format [-Wformat-diag]
2024-08-16T10:34:31.2094222Z   981 |                                                      "'buf' (char[%i])",
2024-08-16T10:34:31.2094941Z       |                                                              ^~~~
2024-08-16T10:34:31.2096219Z ../../gcc/text-art/table.cc:981:69: warning: spurious trailing punctuation sequence '])' in format [-Wformat-diag]
2024-08-16T10:34:31.2097501Z   981 |                                                      "'buf' (char[%i])",
2024-08-16T10:34:31.2098198Z       |                                                                     ^~
2024-08-16T10:34:31.2099178Z make[2]: Leaving directory '/home/runner/work/gccrs/gccrs/gccrs-build/gcc'
```

Reply:
also available in https://github.com/Rust-GCC/gccrs/blob/master/.github/bors_log_expected_warnings#L129:L132

Reply:
Maybe @davidmalcolm is aware of these warnings? :D
They are probably not related to 32bits, so we must have them in the 64bits baseline.

Reply:
@badumbatish I was talking about this line specifically:

```
warning: '*(std::_Vector_base<unsigned int, std::allocator<unsigned int> >*)((char*)&saved + offsetof(Rust::BIR::PatternBindingBuilder::SavedState, Rust::BIR::PatternBindingBuilder::SavedState::regions.tl::optional<Rust::FreeRegions>::<unnamed>.tl::detail::optional_move_assign_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_copy_assign_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_move_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_copy_base<Rust::FreeRegions, false>::<unnamed>.tl::detail::optional_operations_base<Rust::FreeRegions>::<unnamed>.tl::detail::optional_storage_base<Rust::FreeRegions, false>::<unnamed>)).std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_M_impl.std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_Vector_impl::<anonymous>.std::_Vector_base<unsigned int, std::allocator<unsigned int> >::_Vector_impl_data::_M_start' may be used uninitialized [-Wmaybe-uninitialized]
```

that's a gccrs type, that we are using wrong I guess? I agree with you that it can be fixed separately though

Reply:
if it's ok with you i'll resolve the issue Arthur :)

Reply:
but i'm not sure if i should remove the warnings from the expected file or is the current state ok?

Reply:
If you manage to fix the warning you should delete the warning from the expected file, we want to keep warning count as low as possible

Reply:
yes, as @P-E-P said. so keep that PR in its current state but when you fix the warning make sure to remove it from the file :+1: good work!